packages/itmat-ui-react/src/components/lxd/lxd.instance.console.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import { SpiceMainConn, handle_resize } from './spice/src/main';
+import 'xterm/css/xterm.css';
+
+import css from './lxd.module.css';
+
+declare global {
+    // eslint-disable-next-line no-unused-vars
+    interface Window {
+        spice_connection?: SpiceMainConn;
+    }
+}
+
+
+const updateVgaConsoleSize = () => {
+    const spiceScreen = document.getElementById('spice-screen');
+    if (spiceScreen) {
+        // Optionally, you can adjust these offsets based on your UI layout
+        // const offsetTop = spiceScreen.offsetTop;
+        // const height = window.innerHeight - offsetTop;
+        // spiceScreen.style.height = `${height}px`;
+
+        // Set minimum dimensions to ensure the console is usable
+        const minWidth = 800;
+        const minHeight = 600;
+
+        // Calculate available dimensions
+        const availableWidth = Math.max(window.innerWidth, minWidth);
+        const availableHeight = Math.max(window.innerHeight - spiceScreen.offsetTop, minHeight);
+
+        spiceScreen.style.width = `${availableWidth}px`;
+        spiceScreen.style.height = `${availableHeight}px`;
+    }
+};
+
+
+
+
+export const LXDConsole: React.FC<{ instanceName: string }> = ({ instanceName }) => {
+
+    console.log('>>> Enter the LXDConsole instanceName', instanceName );
+
+    const spiceRef = useRef<HTMLDivElement>(null);
+    const [hasStarted, setHasStarted] = useState(false);
+
+    const handleResize = () => {
+        updateVgaConsoleSize();
+        handle_resize();
+    };
+
+    const openVgaConsole = useCallback(async () => {
+
+        // Ensure the size is updated based on the current window size
+        updateVgaConsoleSize();
+
+        const spiceScreen = document.getElementById('spice-screen');
+        const width = spiceScreen ? spiceScreen.clientWidth : 1024; // Provide fallback dimensions
+        const height = spiceScreen ? spiceScreen.clientHeight : 768;
+
+        console.log('spiceScreen width height', width, height);
+
+        const res = await fetch('/lxd/console?c=' + instanceName, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                // width: spiceRef.current?.clientWidth,
+                // height: spiceRef.current?.clientHeight,
+                width: width,
+                height: height,
+                type: 'vga'
+            })
+        });
+
+        if (!res) {
+            return;
+        }
+        const result = await res.json();
+
+        const baseUrl = new URL(window.location.href);
+        baseUrl.protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
+        baseUrl.pathname = '/rtc';
```

Comment:
Probably this is not the correct logic with the new domain specific changed. Or the domain specific things must change to get it done correctly. Eitherway, this probably wouldn't play well with proxies